Gulpfile.js

```diff
@@ -171,6 +170,37 @@ gulp.task("build-check-test", gulp.series(
 gulp.task("build-lint-test", gulp.task("build-check-test"));
 
 // Helper functions:
+function spawnTask(command, done, args)
+{
+    let lCP;
+    if (args !== undefined)
+    {
+        lCP = spawn(command, args, { stdio: "inherit" });
+    } 
+    else
+    {
+        lCP = spawn(command, { stdio: "inherit" });
+    }
+    
+    lCP.on("error", (err) =>
+    {
+        console.error("Failed to start subprocess.");
+        done(err);
+    });
+
+    // catch non-zero exit statii so that CI can understand when task fails
+    lCP.on("close", (code, signal) => {
+        if (code !== 0)
+        {
+            done(new Error(`Task failed with error code ${code}`));
+        }
+        else
+        {
+            done();
+        }
```

Comment:
```suggestion
        done(code !== 0 ? new Error(`Task failed with code: ${code}`) : undefined);
```

If you accept this change, please test it out because some JS libraries will treat `undefined` weirdly. (I think it's just when you do it on an object, e.g. `obj.a = undefined` is diff from `delete(obj.a)` because `Obj.keys(obj)` will catch a even if it's undefined

Reply:
the safest thing to do is just call `done(code)` when code is 0, so that's what I settled on.